homeassistant/components/prusalink/__init__.py

```diff
@@ -79,7 +108,24 @@ async def _async_update_data(self) -> T:
         try:
             async with asyncio.timeout(5):
                 data = await self._fetch_data()
+                # Authentication is working again so we can safely remove it again
+                ir.async_delete_issue(self.hass, DOMAIN, "firmware_5_1_required")
         except InvalidAuth:
+            # We don't know for sure if the firmware is actually outdated
+            # If we hit an InvalidAuth error after the user configured this integration
+            # (where we already checked that credentials are correct)
+            # then its most likely an issue with unsupported firmware.
+            ir.async_create_issue(
+                self.hass,
+                DOMAIN,
+                "firmware_5_1_required",
+                is_fixable=False,
+                learn_more_url="https://help.prusa3d.com/article/firmware-updating-mini-mini_124784",
```

Comment:
Can we somehow identify the model?
This points to the Mini, but there are currently also MK4, MK3.9, and XL.

Reply:
I don't think we can figure out what model it is because a) the API doesn't really provide it (unless the hostname is good enough to figure it out) and b) we cannot reach the API due to outdated firmware anyway.

The good news is that:
* Multi-tool XL initial firmware (4.7) from Jun 27 already supports v1 out of the box.
* MK3.9 initial Firmware (5.0) from Sep 18 already supports v1 out of the box.

Non Multi-tool XL and MK4 initial batch might be shipped with firmware 4.5 or 4.6 which doesn't support v1, but I'm pretty sure that everybody already updated to 4.7.x, 5.0.x or 5.1.x due to many bugs in the initial firmware.

The issue with the Prusa Mini is that it didn't have a firmware update for almost 2 years and only with the very last release it got v1 and digest auth support.

Reply:
Okay, then we should at least mention in the repair issue the required firmware for all models like you have done it here :)

Reply:
As we are mentioning the required firmware per model, we could also include the correct link to the upgrade there. As the link should not be translatable we can pass them via the placeholders.

Similar to: https://github.com/home-assistant/core/blob/f94167a4bbf3e5620fd5ba1e31e6794df90052dd/homeassistant/components/home_plus_control/strings.json#L27

Reply:
I've updated the text and removed the learn_more_url link. It looks like this now:

![image](https://github.com/home-assistant/core/assets/2886913/0265bd5b-725c-4e84-93cc-560723208a05)